Description: Use correct fontconfig syntax for disjunction and don't include multiple elements in test/alias
Author: Iain Lane <iain.lane@canonical.com>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=687043

Index: b/41-arphic-uming.conf
===================================================================
--- a/41-arphic-uming.conf
+++ b/41-arphic-uming.conf
@@ -7,22 +7,52 @@
  -->
 	<alias>
 		<family>AR PL ShanHeiSun Uni</family>
+		<default><family>serif</family></default>
+	</alias>
+	<alias>
 		<family>AR PL ShanHeiSun Uni MBE</family>
+		<default><family>serif</family></default>
+	</alias>
+	<alias>
 		<family>AR PL UMing CN</family>
+		<default><family>serif</family></default>
+	</alias>
+	<alias>
 		<family>AR PL UMing HK</family>
+		<default><family>serif</family></default>
+	</alias>
+	<alias>
 		<family>AR PL UMing TW</family>
+		<default><family>serif</family></default>
+	</alias>
+	<alias>
 		<family>AR PL UMing TW MBE</family>
 		<default><family>serif</family></default>
 	</alias>
 <!--
   Monospace faces
  -->
- 	<alias>
+	<alias>
 		<family>AR PL ShanHeiSun Uni</family>
+		<default><family>monospace</family></default>
+	</alias>
+	<alias>
 		<family>AR PL ShanHeiSun Uni MBE</family>
+		<default><family>monospace</family></default>
+	</alias>
+	<alias>
 		<family>AR PL UMing CN</family>
+		<default><family>monospace</family></default>
+	</alias>
+	<alias>
 		<family>AR PL UMing HK</family>
+		<default><family>monospace</family></default>
+	</alias>
+	<alias>
 		<family>AR PL UMing TW</family>
+		<default><family>monospace</family></default>
+	</alias>
+	<alias>
 		<family>AR PL UMing TW MBE</family>
 		<default><family>monospace</family></default>
 	</alias>
Index: b/64-arphic-uming.conf
===================================================================
--- a/64-arphic-uming.conf
+++ b/64-arphic-uming.conf
@@ -4,6 +4,17 @@
 	<match target="pattern">
 		<test qual="any" name="lang" compare="contains">
 			<string>zh-cn</string>
+		</test>
+		<test qual="any" name="family">
+			<string>serif</string>
+		</test>
+		<edit name="family" mode="prepend" binding="strong">
+			<string>AR PL UMing CN</string>
+		</edit>
+	</match>
+
+	<match target="pattern">
+		<test qual="any" name="lang" compare="contains">
 			<string>zh-sg</string>
 		</test>
 		<test qual="any" name="family">
@@ -17,6 +28,17 @@
 	<match target="pattern">
 		<test qual="any" name="lang" compare="contains">
 			<string>zh-cn</string>
+		</test>
+		<test name="family">
+			<string>AR PL UMing CN</string>
+		</test>
+		<edit name="family" mode="append" binding="strong">
+			<string>DejaVu Serif</string>
+		</edit>
+	</match>
+
+	<match target="pattern">
+		<test qual="any" name="lang" compare="contains">
 			<string>zh-sg</string>
 		</test>
 		<test name="family">
@@ -30,6 +52,17 @@
 	<match target="pattern">
 		<test qual="any" name="lang" compare="contains">
 			<string>zh-hk</string>
+		</test>
+		<test qual="any" name="family">
+			<string>serif</string>
+		</test>
+		<edit name="family" mode="prepend" binding="strong">
+			<string>AR PL UMing HK</string>
+		</edit>
+	</match>
+
+	<match target="pattern">
+		<test qual="any" name="lang" compare="contains">
 			<string>zh-mo</string>
 		</test>
 		<test qual="any" name="family">
@@ -43,6 +76,17 @@
 	<match target="pattern">
 		<test qual="any" name="lang" compare="contains">
 			<string>zh-hk</string>
+		</test>
+		<test name="family">
+			<string>AR PL UMing HK</string>
+		</test>
+		<edit name="family" mode="append" binding="strong">
+			<string>DejaVu Serif</string>
+		</edit>
+	</match>
+
+	<match target="pattern">
+		<test qual="any" name="lang" compare="contains">
 			<string>zh-mo</string>
 		</test>
 		<test name="family">
Index: b/25-arphic-uming-bitmaps.conf
===================================================================
--- a/25-arphic-uming-bitmaps.conf
+++ b/25-arphic-uming-bitmaps.conf
@@ -5,25 +5,65 @@
 	<match target="font">
 	  <test name="family">
 	  <string>AR PL UMing CN</string>
+	  </test>
+       	  <edit name="antialias"><bool>false</bool></edit>
+    	  <edit name="hinting"><bool>true</bool></edit>
+          <edit name="autohint"><bool>false</bool></edit>
+	</match>
+	<match target="font">
+	  <test name="family">
 	  <string>AR PL UMing HK</string>
+	  </test>
+       	  <edit name="antialias"><bool>false</bool></edit>
+    	  <edit name="hinting"><bool>true</bool></edit>
+          <edit name="autohint"><bool>false</bool></edit>
+	</match>
+	<match target="font">
+	  <test name="family">
 	  <string>AR PL UMing TW</string>
+	  </test>
+       	  <edit name="antialias"><bool>false</bool></edit>
+    	  <edit name="hinting"><bool>true</bool></edit>
+          <edit name="autohint"><bool>false</bool></edit>
+	</match>
+	<match target="font">
+	  <test name="family">
 	  <string>AR PL UMing TW MBE</string>
 	  </test>
        	  <edit name="antialias"><bool>false</bool></edit>
     	  <edit name="hinting"><bool>true</bool></edit>
           <edit name="autohint"><bool>false</bool></edit>
 	</match>
-
 	<match target="font">
 	  <test name="family">
 	  <string>AR PL UMing CN</string>
+	  </test>
+      	  <test name="pixelsize" compare="more_eq"><int>17</int></test>
+	  <edit name="antialias" mode="assign"><bool>true</bool></edit>
+	  <edit name="hinting" mode="assign"><bool>true</bool></edit>
+	</match>
+	<match target="font">
+	  <test name="family">
 	  <string>AR PL UMing HK</string>
+	  </test>
+      	  <test name="pixelsize" compare="more_eq"><int>17</int></test>
+	  <edit name="antialias" mode="assign"><bool>true</bool></edit>
+	  <edit name="hinting" mode="assign"><bool>true</bool></edit>
+	</match>
+	<match target="font">
+	  <test name="family">
 	  <string>AR PL UMing TW</string>
+	  </test>
+      	  <test name="pixelsize" compare="more_eq"><int>17</int></test>
+	  <edit name="antialias" mode="assign"><bool>true</bool></edit>
+	  <edit name="hinting" mode="assign"><bool>true</bool></edit>
+	</match>
+	<match target="font">
+	  <test name="family">
 	  <string>AR PL UMing TW MBE</string>
 	  </test>
       	  <test name="pixelsize" compare="more_eq"><int>17</int></test>
 	  <edit name="antialias" mode="assign"><bool>true</bool></edit>
 	  <edit name="hinting" mode="assign"><bool>true</bool></edit>
 	</match>
-
 </fontconfig>
